corgi/collectors/models.py

```diff
@@ -43,21 +43,45 @@ def __str__(self):
         return f"{self.nvr}"
 
 
-class CollectorComposeSRPM(models.Model):
+class CollectorSRPM(models.Model):
 
     build_id = models.IntegerField(primary_key=True)
 
     def __str__(self):
         return f"{self.build_id}"
 
 
-class CollectorComposeRPM(models.Model):
+class CollectorRPM(models.Model):
     """Not every RPM built as part of a SRPM is included in a module
     Which is why we don't directly related SRPM builds to product_streams using relations table"""
 
     nvra = models.TextField(unique=True)
-    rhel_module = models.ManyToManyField(CollectorComposeRhelModule)
-    srpm = models.ForeignKey(CollectorComposeSRPM, on_delete=models.CASCADE, related_name="rpms")
+    rhel_module = models.ManyToManyField(CollectorRhelModule)
+    srpm = models.ForeignKey(CollectorSRPM, on_delete=models.CASCADE, related_name="rpms")
 
     def __str__(self):
         return f"{self.nvra}"
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                name="unique_collectorrpms",
+                fields=("nvra", "srpm"),
+            ),
+        ]
+
+
+class CollectorRPMRepository(models.Model):
+    name = models.CharField(unique=True, max_length=200)
+    content_set = models.CharField(max_length=200)
+
+    def __str__(self):
+        return f"{self.name}"
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                name="unique_collectorrpmrepos",
+                fields=("name", "content_set"),
+            ),
```

Comment:
As above - since name is always unique, forcing name + content_set to be unique is redundant.

Reply:
sorry I missed this one on the last pass. Good idea, I will change as you suggest.